### PR TITLE
[WIP] logging refactor

### PR DESCRIPTION
**Pull request by huanggze**
Signed-off-by: huanggze <loganhuang@yunify.com>

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>

/kind api-change
> /kind bug

/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
- Move to KubeSphere 3.0
- Remove APIs for Fluent Bit CRD
- Add testing
- Finish code decoupling work
- Improve docs and code comments

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for reviewers**:
```
```

**Additional documentation, usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by huanggze** [comment]
This pr is for 3.0 and mainly reviewed by @zryfish 

/assign @benjaminhuo

File: pkg/kapis/logging/v1alpha2/handler.go [hunk 1]
```diff
@@ -1 +1,190 @@
...
+
+func (h handler) handleClusterQuery(req *restful.Request, resp *restful.Response) {
+	Get(req, LevelCluster, h.Client, h.LoggingOperator, resp)
+}
+
+func (h handler) handleContainerQuery(req *restful.Request, resp *restful.Response) {
+	Get(req, LevelContainer, h.Client, h.LoggingOperator, resp)
+}
+
+func Get(req *restful.Request, lvl int, client k8s.Client, loggingOperator logging.LoggingOperator, resp *restful.Response) {
```
Comment (zryfish) [inline pkg/kapis/logging/v1alpha2/handler.go]: If `Get` only called inside handler, then it's unnecessary to make `Get` an exported function. Also I think it's better to define `Get` a member method of handler

File: pkg/kapis/logging/v1alpha2/handler.go [hunk 2]
```diff
@@ -1 +1,190 @@
...
+		}
+		res, err := loggingOperator.SearchLogs(sf, from, size, order)
+		if err != nil {
+			api.HandleInternalError(resp, err)
+		}
+		resp.WriteAsJson(res)
+	}
+}
+
+func NewSearchFilter(client k8s.Client, level int, req *restful.Request) (bool, loggingclient.SearchFilter, error) {
```
Comment (zryfish) [inline pkg/kapis/logging/v1alpha2/handler.go]: same here, is it necessary to define `NewSearchFilter` an exported function

File: pkg/kapis/tenant/v1alpha2/register.go [hunk 3]
```diff
@@ -173,8 +173,8 @@ func AddToContainer(c *restful.Container, k8sClient k8s.Client, db *mysql.Databa
 		Param(ws.QueryParameter("from", "The offset from the result set. This field returns query results from the specified offset. It requires **operation** is set to query. Defaults to 0 (i.e. from the beginning of the result set).").DataType("integer").DefaultValue("0").Required(false)).
 		Param(ws.QueryParameter("size", "Size of result to return. It requires **operation** is set to query. Defaults to 10 (i.e. 10 log records).").DataType("integer").DefaultValue("10").Required(false)).
 		Metadata(restfulspec.KeyOpenAPITags, []string{constants.TenantResourcesTag}).
-		Writes(v1alpha2.Response{}).
-		Returns(http.StatusOK, api.StatusOK, v1alpha2.Response{})).
+		Writes(logging.APIResponse{}).
```
Comment (zryfish) [inline pkg/kapis/tenant/v1alpha2/register.go]: different logging module from pkg/kapis/logging/v1alpha2/register.go

File: pkg/kapis/logging/v1alpha2/handler.go [hunk 4]
```diff
@@ -2,73 +2,73 @@ package v1alpha2
...
+func (h handler) handleClusterQuery(req *restful.Request, resp *restful.Response) {
+	Get(req, LevelCluster, h.Client, h.LoggingOperator, resp)
+}
+
+func (h handler) handleContainerQuery(req *restful.Request, resp *restful.Response) {
+	Get(req, LevelContainer, h.Client, h.LoggingOperator, resp)
+}
+
+func Get(req *restful.Request, lvl int, client k8s.Client, loggingOperator logging.LoggingOperator, resp *restful.Response) {
 	typ := req.QueryParameter("type")
```
[the remark this answers is not in the document]
Comment (huanggze) [reply pkg/kapis/logging/v1alpha2/handler.go]: type is a keyword. Many other projects use typ as a var name

[the remark this answers is not in the document]
Comment (huanggze) [reply pkg/kapis/logging/v1alpha2/handler.go]: type is a keyword. Many other projects use typ as a var name

File: pkg/models/logging/helper.go [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+		for _, ns := range nsList.Items {
+			namespaces = append(namespaces, ns.Name)
+		}
+	} else {
+		for _, ns := range nsList.Items {
+			if stringutils.StringIn(ns.Name, nsFilter) ||
+				stringutils.StringIn(ns.Annotations[constants.WorkspaceLabelKey], wsFilter) ||
+				containsIn(ns.Name, nsSearch) ||
+				containsIn(ns.Annotations[constants.WorkspaceLabelKey], wsSearch) {
+				namespaces = append(namespaces, ns.Name)
```
[the remark this answers is not in the document]
Comment (benjaminhuo) [reply pkg/models/logging/helper.go]: The new changes looks good for this. 
It's more efficient than creating a map in another separate function

File: pkg/kapis/logging/v1alpha2/handler.go (hunk 1 above)
Comment (huanggze) [reply pkg/kapis/logging/v1alpha2/handler.go]: the reason is that tenant api depends this, i have suggested @wansir moving codes for multi-tenant logging to logging module

File: pkg/kapis/logging/v1alpha2/handler.go (hunk 2 above)
Comment (huanggze) [reply pkg/kapis/logging/v1alpha2/handler.go]: done

**Pull request merged**